lib/internal/Magento/Framework/Setup/Declaration/Schema/Db/MySQL/DbSchemaReader.php

```diff
@@ -225,7 +225,8 @@ public function readTables($resource)
                 ['TABLE_NAME']
             )
             ->where('TABLE_SCHEMA = ?', $dbName)
-            ->where('TABLE_TYPE = ?', self::MYSQL_TABLE_TYPE);
+            ->where('TABLE_TYPE = ?', self::MYSQL_TABLE_TYPE)
+            ->where('TABLE_NAME LIKE ?', $this->resourceConnection->getTablePrefix() . '%');
```

Comment:
Should wildcard characters (like `_`) be escaped from `$this->resourceConnection->getTablePrefix()`?

Reply:
@fredden good point. And yes, it should be escaped